### PR TITLE
fix: destroy tray on current tick

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Resolves https://github.com/electron/electron/issues/17622.

This code was originally added in https://github.com/electron/electron/pull/6448 to handle an edge case crash in 10.9, and we no longer support 10.9 and therefore no longer need to account for this case.

It addressed the crash, but also created a race condition whereby when a new tray is created the old tray's destroy wouldn't have been fully completed and therefore a new one would be spawned. This fixes that by destroying the tray on the current tick once more. 

cc @zcbenz @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where tray.destroy was not working properly on some linux distros.

### CODE REVIEW

**Comment by codebytere** [comment]
@deepak1556 i added some specs but if there are more specifics that i should test please let me know!

**Comment by MarshallOfSound** [comment]
Can we add a test for the original issue?  (Calling `app.quit()` inside menu even handler).  Not sure if this is possible but would be good if it is 👍

File: atom/browser/api/atom_api_tray.cc [hunk 1]
```diff
@@ -60,11 +60,7 @@ Tray::Tray(v8::Isolate* isolate,
   InitWith(isolate, wrapper);
 }
 
-Tray::~Tray() {
-  // Destroy the native tray in next tick.
-  base::ThreadTaskRunnerHandle::Get()->DeleteSoon(FROM_HERE,
-                                                  tray_icon_.release());
-}
+Tray::~Tray() {}
```
Comment (miniak) [inline atom/browser/api/atom_api_tray.cc]: you can use fancy C++11 `Tray::~Tray() = default;` :)

**Comment by codebytere** [comment]
@MarshallOfSound i don't think that's possible sadly :( 
The original issue didn't call `app.quit()` i don't believe?

**Comment by codebytere** [comment]
Mac failure is a flake: merging!

**Pull request merged**